### DISCUSSION TITLE
S3 data replication

### DISCUSSION THREAD

**Discussion by EmilHiVa** [grafana/mimir#5899] [Help and support] [answered]
Hi.

Is it possible to have Mimir duplicate writes to more than one S3 target? This is for data availability purpose.

I have been looking at store-gateway configuration, but I'm not sure I fully understand the architecture.
Will a replication factor of 2, with zone-awerenes and different block_storage sections be enough?

Thanks you in advance and please excuse my limited understanding of Mimir - I'm a usually a full time storage guy.

**Comment by dimitarvdimitrov** [comment, marked as answer]
it's currently not possible to do that. The ingesters and the compactor are the components which write to the S3 bucket and they only support writing to a single endpoint. The store-gateways also need a unified view of the bucket (i.e. you cannot configure different availability zones differently). 

I think to achieve this you'd need to take care of the bucket sync in the background and reconfigure which bucket mimir uses when one is unavailable.

**Reply by EmilHiVa** [reply]
Thank you very much for your explanation.

I'll pursue a different way to archive the desired fault tolerance then.
